unitary/alpha/qudit_gates.py

```diff
@@ -34,28 +34,32 @@ def __init__(
         self.dimension = dimension
         self.source_state = source_state
         self.destination_state = destination_state
+        if self.source_state >= self.dimension:
+            raise ValueError("Source state must be smaller than dimension.")
+        if self.destination_state >= self.dimension:
+            raise ValueError("Destination state must be smaller than dimension.")
 
     def _qid_shape_(self):
         return (self.dimension,)
 
     def _unitary_(self):
-        arr = np.zeros((self.dimension, self.dimension))
-        arr[self.source_state, self.destination_state] = 1
-        arr[self.destination_state, self.source_state] = 1
-        for i in range(self.dimension):
-            if i != self.source_state and i != self.destination_state:
-                arr[i, i] = 1
+        arr = np.eye(self.dimension)
```

Comment:
These are great changes, but let's split them out into a different PR since they are not relevant to the creation of a generalized Hadamard.

Reply:
Thanks Doug. They are split out and will be shared in another PR.